### ISSUE TITLE
Provide css emmet abbr suggestions when inside style attribute in non html files

### ISSUE THREAD

**Issue by Kcko** [Microsoft/vscode#45275]
See short video please https://goo.gl/5uGtFB (suggestion for emmet not works in latte file with inline style. In html file it works, settings. Emmett settings are complete in editor settings.

**Comment by ramya-rao-a** [comment]
Do you use any extensions for the latte files? What language mode are these files mapped to? What language do you see in the bottom right corner of the editor?

**Comment by Kcko** [comment]
Hi,
at first thank you for reply:
now to our problem:

1) I see latte in the right bottom corner (https://goo.gl/wCsGYQ)
2) I have installed extension by martin kase Nette + Latte (https://marketplace.visualstudio.com/items?itemName=Kasik96.latte), but when i disable it, so it has no effect, still not works.
3) See short video (in html file/lang emmet at inline style works, in latte not) https://goo.gl/h4X3Q5

Thank you for your willingness to solve the problem.

**Comment by ramya-rao-a** [comment]
For filetypes that do not have emmet support by default, you can use the setting `emmet.includeLanguages` setting get emmet completions. 

Css Emmet completions in the inline style attribute can only be provided in html files by parsing the html files to find the right position for the inline style attributes.

In non html files, this is not possible.

One option for you is to add the below setting
```
    "files.associations": {
        "*.latte": "html"
    }
```

This will ensure all latte files get treated as html, so you will get all the features that you get in html files

**Comment by Kcko** [comment]
Thx, works!

But dont understand what directive emme.includeLanguages doing.
https://goo.gl/93zFCC (second directive (by you), works).

**Comment by ramya-rao-a** [comment]
Mapping `latte` to `html` in the `emmet.includeLanguages` setting lets you do the below:
- Use completions for **html** emmet abbreviations in the auto-completion list
- Use the `Emmet: Expand Abbreviation` command to expand the **html** emmet abbreviations
- If you set `emmet.triggerExpansionOnTab` to `true`, then it lets you expand **html** emmet abrbeviations by pressing the tab key.

What it does **not** do is parse the file to find inline styles to provide css emmet completions.

Your expectation here is to get css emmet completions when inside the inline style attribute in an html like file. This is only possible if there is a parser that is able to parse the latte file to find inline styles. By mapping `latte` to `html` in the `files.associations` setting, the html parser gets triggered and does the needful.

In short, mapping `latte` to `html` in `emmet.includeLanguages` lets you expand html emmet abbreviations. Mapping `latte` to `html` in `files.associations` treats the latte files as html files giving you ALL the features that you see in a normal html file.

Hope this clarifies everything

**Comment by ramya-rao-a** [comment]
I'll re-open this issue to investigate if there is anything more we can do from our side to include css emmet completions in the inline style attribute in non html files

**Comment by Kcko** [comment]
Hi @ramya-rao-a , thx for explanation

With mapping latte to html in files.associations  I will lose all Latte properties and behavior so I can not use it.

Only solutions is creating Latte syntatic parser which can have same behavior like a html syntatic parser (your words).

**Comment by ramya-rao-a** [comment]
> Only solutions is creating Latte syntatic parser which can have same behavior like a html syntatic parser (your words)

Yes, thats the ideal solution.

But I think we can do a best guess by traversing left from the current position to determine if the current position is inside a style attribute or not and then provide the emmet css completions

**Comment by Kcko** [comment]
Not so important, inline styles are often not used. Thank you for your consultation. We can close now ;-)

**Comment by ramya-rao-a** [comment]
> We can close now

Not at all :) Now I am all excited to get that feature. It may not be used often, but it is do-able. It makes the feature "complete". In the latest release of VS Code, you can get css emmet completions inside style attributes in html files. Its only fair that we extend this to non html files that have html markup.

**Comment by Kcko** [comment]
I will see roadmap or changelog :-) and i hope i see this feature soon. 

Im programmer  & coder,  im flew from SublimeText ,and im worried about other things than this.
VSC doesnt invoke used classes and identifier in css / scss when I'm in HTML and vice versa. (Non existing intelisense between CSS/SCSS/LESS  and HTML, this is a real problem. But it will not be your circuit probably.

**Comment by ramya-rao-a** [comment]
@Kcko Check existing feature requests for [html](https://github.com/Microsoft/vscode/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request+label%3Ahtml) and [css/scss/less](https://github.com/Microsoft/vscode/issues?q=is%3Aopen+is%3Aissue+label%3Acss-less-sass+label%3Afeature-request). There must be some already that cover what you are looking for. Upvote those feature requests  and add your comments.

**Comment by Kcko** [comment]
@ramya-rao-a  Unfortunately nothing. I tried a few extensions, but without positive results.
Moreover I think, that could be a native property of editor. Whats your opinion?

If you try a PHPStorm, you will see a luxury behavior and intelisense.

**Comment by ramya-rao-a** [comment]
@Kcko We have feature request tracking what you need. See https://github.com/Microsoft/vscode/issues/28442. Upvote and add your comments there.

**Comment by Kcko** [comment]
Thx for link, i will make some vote or add comment :-), we really can close  this topic.

**Comment by ramya-rao-a** [comment]
Alright
